gnss_ndt_selector/src/localizer_node.cpp

```diff
@@ -71,6 +79,53 @@ namespace localizer
 		}
 	}
 
+	bool Localizer::spinCallback()
+	{
+		// once initialized check if sensors are timeout
+		if (gnss_initialized_)
+		{
+			// check if timeout
+			if (gnss_operational_ && (ros::Time::now() - gnss_last_received_).toSec() > double(gnss_time_out_ / 1000))
+			{
+				ROS_WARN_STREAM("GNSS has timed out since:" << gnss_last_received_);
+				gnss_operational_ = false;
+			}
+			// try to switch to ndt if gnss not operational
+			if (!gnss_operational_ && ndt_operational_)
+			{
+				ROS_WARN_STREAM("Since GNSS has timed out, switching to NDT");
+				localization_mode_ = LocalizerMode::NDT;
+			}
+			else if (!gnss_operational_ && !ndt_operational_) //if both not operational although initialized
+			{
+				ROS_ERROR_STREAM("Both NDT and GNSS have timed out! Please take manual control!");
+				return false;
```

Comment:
I don't think its nessasary to kill the node (which is what returning false will do). If operational state is restored that's not a bad thing.